File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/data/items/Item.scala





















// Generated on 05/03/2013 10:35:06
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait Item extends TypeDeserializer {
  val typeId = 7

  protected trait Type extends BaseType with Serializable {
    val typeId = Item.typeId


    def serialize(buf: Buffer) {
    }

  }

  protected def create(): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {

    create()
  }

}

object Item extends Item {
  case class TypeImpl() extends Type
  override def create() = TypeImpl()

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply() = create()

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/Item.java


// Generated on 05/08/2013 19:38:02
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class Item extends NetworkType {
    public static final short TYPE_ID = 7;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    
    public Item() { }
    
    
    @Override
    public void serialize(Buffer buf) {
    }
    
    @Override
    public void deserialize(Buffer buf) {
    }
    
}
